src/components/NcBlurHash/NcBlurHash.vue

```diff
@@ -0,0 +1,115 @@
+<!--
+  - SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+<docs>
+## BlurHash
+
+A [blur hash](https://blurha.sh/) is a very compact representation of an image,
+that can be used as a placeholder until the image was fully loaded.
+
+```vue
+	<template>
+		<NcButton type="tertiary" @click="showImage = !showImage">
+			<div style="display: flex; flex-direction: column;">
+				<img v-show="showImage"
+					alt=""
+					class="button-image"
+					src="styleguide/assets/img/favicon.svg">
+				<NcBlurHash v-show="!showImage"
+					class="button-image"
+					:hash="blurHash" />
+				Toggle BlurHash
+			</div>
+		</NcButton>
+	</template>
+	<script>
+	export default {
+		data() {
+			return {
+				showImage: false,
+				blurHash: 'M8CR]OkDD%kD9ZtRayofaykC00ay$_ay~T',
+			}
+		},
+	}
+	</script>
+	<style>
+		.button-image {
+			width: 150px;
+			height: 150px;
+			border-radius: 24px;
+		}
+	</style>
+```
+</docs>
+
+<script setup>
+import { useResizeObserver } from '@vueuse/core'
+import { decode } from 'blurhash'
+import { onMounted, ref, watch } from 'vue'
+import { logger } from '../../utils/logger.ts'
+
+const props = defineProps({
+	/**
+	 * The blur hash value to use.
+	 */
+	hash: {
+		required: true,
+		type: String,
+	},
+
+	/**
+	 * This is normally not needed, but if this blur hash is not only intended
+	 * for decorative purpose, descriptive text should be passed for accessibility.
+	 */
+	label: {
+		type: String,
+		default: ''
+	}
+})
+
+const canvas = ref()
+
+// Draw initial version on mounted
+onMounted(drawBlurHash)
+
+// On resize we redraw the BlurHash
+useResizeObserver(canvas, drawBlurHash)
```

Comment:
I'm not sure if we need to recalculate blurhash on resize. Isn't it scaling fine (with CSS properties)?

Reply:
done